test/e2e/common.go

```diff
@@ -359,3 +363,106 @@ func IsBootedFromDisk(client *ssh.Client) (bool, error) {
 
 	return isDisk, nil
 }
+
+func DeploymentRolledOut(ctx context.Context, clusterProxy framework.ClusterProxy, name string, namespace string, desiredGeneration int64) bool {
+	clientSet := clusterProxy.GetClientSet()
+	deploy, err := clientSet.AppsV1().Deployments(namespace).Get(ctx, name, metav1.GetOptions{})
+	Expect(err).To(BeNil())
+	if deploy != nil {
+		// When the number of replicas is equal to the number of available and updated
+		// replicas, we know that only "new" pods are running. When we also
+		// have the desired number of replicas and a new enough generation, we
+		// know that the rollout is complete.
+		return (deploy.Status.UpdatedReplicas == *deploy.Spec.Replicas) &&
+			(deploy.Status.AvailableReplicas == *deploy.Spec.Replicas) &&
+			(deploy.Status.Replicas == *deploy.Spec.Replicas) &&
+			(deploy.Status.ObservedGeneration >= desiredGeneration)
+	}
+	return false
+}
+
+func CopyFile(src, dst string) error {
+	out, err := os.Create(filepath.Clean(dst))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = out.Close()
+	}()
+
+	in, err := os.Open(filepath.Clean(src))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = in.Close()
+	}()
+
+	_, err = io.Copy(out, in)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type InstallIronicBMOInput struct {
+	Kustomization       string
+	IronicEnvFile       string
+	ClusterProxy        framework.ClusterProxy
+	WaitForDeployment   bool
+	WatchDeploymentLogs bool
+	DeploymentName      string
+	DeployNS            string
+	LogPath             string
+	WaitIntervals       []interface{}
+}
+
+func InstallIronicBMO(ctx context.Context, input *InstallIronicBMOInput) error {
+	var err error
+	kustomization := input.Kustomization
+	ironicEnvFile := input.IronicEnvFile
+	clusterProxy := input.ClusterProxy
+	if ironicEnvFile != "" {
+		dst := filepath.Join(kustomization, "ironic.env")
+		err = CopyFile(ironicEnvFile, dst)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Why? Just let the file be in the kustomization so it can be applied directly.

Reply:
It's related to the ip address needed by the cluster. Please see my more detailed explanation in another reply.